### PR TITLE
WIP: duplicate columns

### PR DESCRIPTION
**Pull request by pratapvardhan**

### CODE REVIEW

**Comment by pratapvardhan** [comment]
@naveenreddymanukonda -- Sorry, I didn't mean to merge it into dev. This was meant as a placeholder to review..

File: dataaudit.py [hunk 1]
```diff
@@ -157,6 +190,131 @@ def count_categorical_outliers(series):
...
+                    jv_series = vs.iloc[:,j].fillna(0)
+                else:
+                    iv_series = vs.iloc[:,i]
+                    jv_series = vs.iloc[:,j]
+                iv = iv_series.tolist()
+                jv = jv_series.tolist()
+                if np.array_equiv(iv, jv):
+                    dups.append(cs[i])
+                    break
+    return dups
```
Comment (pratapvardhan) [inline dataaudit.py:270]: This seems a bit verbose. Does this help solve the utility?

```
In [118]: for i, ac in enumerate(df):
     ...:     for j, bc in enumerate(df):
     ...:         if i >= j:
     ...:             continue
     ...:         if df[ac].equals(df[bc]):
     ...:             print(ac, bc)
     ...:
     ...:
product product.1
sales sales.1
growth growth.1
```

File: dataaudit.py [hunk 2]
```diff
@@ -157,6 +190,131 @@ def count_categorical_outliers(series):
...
+    '''
+    order_id_continous_columns = []
+    for column in data:
+        s_data = data[column]
+        if s_data.dtypes == np.float64 or s_data.dtypes == np.int64:
+            s_data.fillna(0, inplace=True)
+            s_data_diff = s_data.diff().reset_index(drop=True)
+            if len(s_data_diff.unique()) == 2:
+                order_id_continous_columns.append(column)
+    if len(order_id_continous_columns) > 0:
```
Comment (pratapvardhan) [inline dataaudit.py:285]: How about

```
def is_plus_one(series):
    return pd.api.types.is_numeric_dtype(series) and series.diff().ne(1).sum() == 1

[col for col in df if is_plus_one(df[col])]
```

File: dataaudit.py [hunk 3]
```diff
@@ -157,6 +190,131 @@ def count_categorical_outliers(series):
...
+    '''
+    Given dataframe check primekey unique.
+    '''
+    primary_key_unique_columns = []
+    for column in data:
+        s_data = data[column]
+        if not (s_data.isnull().values.any()):
+            if len(s_data) == len(s_data.unique()):
+                primary_key_unique_columns.append(column)
+    if len(primary_key_unique_columns) > 0:
```
Comment (pratapvardhan) [inline dataaudit.py:303]: Have considered? `pd.Series.is_unique`

```
[c for c in df if df[c].is_unique]
```

File: dataaudit.py (hunk 2 above)
Comment (naveenreddymanukonda) [reply dataaudit.py:285]: @pratapvardhan difference of the sum may 1 or `n` but it should be consistent through out the column.
Example:

order_id
--
100
200
300
400

File: dataaudit.py (hunk 3 above)
Comment (naveenreddymanukonda) [reply dataaudit.py:303]: @pratapvardhan these works thanks :)

File: dataaudit.py (hunk 2 above)
Comment (pratapvardhan) [reply dataaudit.py:285]: `series.diff().nunique() == 1` then?

Comment (naveenreddymanukonda) [reply dataaudit.py:285]: these wortks @pratapvardhan thanks :)

File: dataaudit.py (hunk 1 above)
Comment (naveenreddymanukonda) [reply dataaudit.py:270]: @pratapvardhan thanks :)

**Pull request merged**